File: python/primefactors.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

""" Python program to calculate prime factors of a given number.

@File = primefactors.py
@Date = 2023-03-29
@Author = Jannis Metrikat
@Version = 1.0

"""

import sys


# calculate prime factors of a given number
def primefactors(n: int) -> None:
    found_factor = False

    for i in range(2, n):
        is_prime = True

        for j in range(2, i):
            if i % j == 0:
                is_prime = False
                break

        if n % i == 0 and is_prime:
            print(i)
            found_factor = True

    if not found_factor:
        print(str(n) + " is prime.")

    return


# main entry point
if len(sys.argv) != 2:
    print("Usage: python primefactors.py <number>")
    sys.exit(1)

if int(sys.argv[1]) < 2:
    print("Number must be greater than 1.")
    sys.exit(1)

primefactors(int(sys.argv[1]))

File: java/Primefactors.java
/**
 * Java program to calculate prime factors of a given number.
 *
 * @File = Primefactors.java
 * @Date = 2023-04-13
 * @Author = Jannis Metrikat
 * @Version = 1.0
 */

public class Primefactors {

    /**
     * calculate prime factors of a given number
     */
    public static void primeFactors(int n) {
        boolean foundFactor = false;

        for (int i = 2; i < n ; i++) {
            boolean isPrime = true;

            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }

            if (n % i == 0 && isPrime) {
                System.out.println(i);
                foundFactor = true;
            }
        }

        if (!foundFactor) {
            System.out.println(n + " is prime.");
        }
    }

    /**
     * main method
     */
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java Primes <number>");
            System.exit(1);
        }

        int n = Integer.parseInt(args[0]);
        if (n < 2) {
            System.out.println("Number must be greater than 1.");
            System.exit(1);
        }

        primeFactors(n);
        return;
    }
}
